pennylane/devices/qubit/simulate.py

```diff
@@ -300,22 +300,29 @@ def simulate_native_mcm(
             results.append(simulate_native_mcm(aux_circuit, rng, prng_key, debugger, interface))
         return tuple(results)
     aux_circuit = init_auxiliary_circuit(circuit)
-    all_shot_meas, list_mcm_values_dict = None, []
+    all_shot_meas, list_mcm_values_dict, valid_shots = None, [], 0
     for _ in range(circuit.shots.total_shots):
```

Comment:
i suppose this isn't related to this PR strictly, but does this account for shot-vectors correctly? eg. if a user uses `shots=[1000, 2000]`, will they get 2 results (one from the first 1000 shots, discarding the invalid ones, and another from the final 2000 shots)?

Reply:
That's a good question. Yes, I thought that's what shot-vectors do, but I may be wrong there. I think what could happen is that a user will get a warning for all shots. Or maybe a warning for some items in the shot-vector and not others. Is that what you're wondering? Maybe I can add a test for that, if so.

Reply:
never mind, ignore me! I was wondering where the nested for-loop is for the case of partitioned shots, but I scrolled up and I see that you factored it out into a nested call 3 lines above this one :p lgtm